### PR TITLE
fix(Datagrid): implemented nestedRow dynamic width

### LINKED ISSUE #4972: Overlapping columns in nested rows — Reduce width instead of margin in first column of nested row
### Package

Carbon for IBM Products

### Description

Currently, to account for the first column of a nested row being smaller due to it being indented, the `margin-right` property is reduced by 50px (and more for further nesting). This however doesn't account for text/content for the column being more than the available space and leads to content overlapping with the next column (This seems to especially be an issue on smaller screens.

Switching to reducing the width of the cell instead would stop this overlapping and allow text-breaking etc. To do this I think there would need to be a way to get the width of this cell (either by passing the value set on the parent column, or getting the width of the cell itself, and reducing this by the same amount as is currently being set by `margin-right`. 

image

image

### Component(s) impacted

Datagrid

### Browser

Firefox

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

v.2.35.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Product/offering

Cloud Pak For Integration (Platform UI)

### CodeSandbox or Stackblitz example

https://ibm-products.carbondesignsystem.com/?path=/docs/ibm-products-components-datagrid-nestedrows--docs

### Steps to reproduce the issue (if applicable)

- Create Nested rows dataGrid
- Observe column overlap on small screen size or in dev tools

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by devadula-nandan**
Closes #4972 

made the width dynamically change, instead of giving margin-right = -50 to fix overlapping issue.
also if any cell is resized below 50px (minimum width), the difference is added to margin right to fix the layout of the row

#### What did you change?
packages/ibm-products/src/components/Datagrid/useNestedRows.js

#### How did you test and verify your work?
storybook

### CODE REVIEW

File: packages/ibm-products/src/components/Datagrid/useNestedRows.js [hunk 1]
```diff
@@ -64,11 +64,21 @@ const useNestedRows = (hooks) => {
         props,
         {
           style: {
-            marginRight: `${
+            marginRight:
+              isFirstCell &&
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/Datagrid/useNestedRows.js:68]: Hey ,
Looks good and seems to be working.
Can you add some comments regarding the logic and figures used here for calculation?

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/Datagrid/useNestedRows.js (hunk 1 above)
Comment (devadula-nandan) [reply packages/ibm-products/src/components/Datagrid/useNestedRows.js:69]: done some refactors for readability, and added comments

**Comment by devadula-nandan** [comment]
@amal-k-joy added comments as per discussion

**Review by amal-k-joy — APPROVED** [review]
Good work

**Review by IgnacioBecerra — APPROVED** [review]
LGTM!

**Pull request merged**